File: 151004/Sakovsky/src/entities/Marker.ts
import { Column, Entity, JoinTable, ManyToMany, PrimaryGeneratedColumn } from "typeorm";
import { Tweet } from "./Tweet";

@Entity({name: 'tbl_markers'})
export class Marker {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    name: string;

    @ManyToMany(() => Tweet)
    @JoinTable({ name: 'tbl_tweets_markers' })
    tweets: Tweet[];
}

File: 151004/Stahnov/kafka/publicator/src/main/java/org/education/bean/Marker.java
package org.education.bean;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@NoArgsConstructor
@Setter
@Getter
@Entity(name = "tbl_marker")
public class Marker {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int id;
    String name;

    @ManyToMany
    @JoinTable(
            name = "TweetMarker",
            joinColumns = @JoinColumn(name = "tweetId", referencedColumnName = "id"),
            inverseJoinColumns = @JoinColumn(name = "markerId", referencedColumnName = "id")
    )
    List<Tweet> tweets;
}